kratos/solving_strategies/schemes/residual_based_bossak_velocity_scheme.h

```diff
@@ -0,0 +1,749 @@
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "solving_strategies/schemes/scheme.h"
+#include "utilities/derivatives_extension.h"
+#include "utilities/indirect_scalar.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/// A scheme for dynamic adjoint equations, using Bossak time integration.
+/**
+ * It can be used for either first- or second-order time derivatives. Elements
+ * and conditions must provide a specialization of AdjointExtensions via their
+ * data value container, which allows the scheme to operate independently of
+ * the variable arrangements in the element or condition.
+ */
+template <class TSparseSpace, class TDenseSpace>
+class ResidualBasedBossakVelocityScheme : public Scheme<TSparseSpace, TDenseSpace>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION(ResidualBasedBossakVelocityScheme);
+
+    typedef Scheme<TSparseSpace, TDenseSpace> BaseType;
+
+    typedef typename BaseType::TSystemMatrixType SystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType SystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef std::vector<Dof<double>::Pointer> DofsVectorType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+
+    ResidualBasedBossakVelocityScheme(const double AlphaBossak,
+                                      const bool UpdateAcceleration = true,
+                                      const bool UpdateDisplacement = true)
+    {
+        mAlphaBossak = AlphaBossak;
+        mUpdateAcceleration = UpdateAcceleration;
+        mUpdateDisplacement = UpdateDisplacement;
+
+        KRATOS_INFO("ResidualBasedBossakVelocityScheme")
+            << " Using bossak velocity scheme with alpha_bossak = " << std::scientific
+            << mAlphaBossak << " [UpdateAcceleration: " << mUpdateAcceleration
+            << ", UpdateDisplacement: " << mUpdateDisplacement << "]\n";
+
+        // Allocate auxiliary memory.
+        int num_threads = OpenMPUtils::GetNumThreads();
+
+        mMassMatrix.resize(num_threads);
+        mDampingMatrix.resize(num_threads);
+        mValuesVector.resize(num_threads);
+        mSecondDerivativeValuesVector.resize(num_threads);
+        mSecondDerivativeValuesVectorOld.resize(num_threads);
+    }
+
+    ResidualBasedBossakVelocityScheme(Parameters& rParameters)
+    {
+        Parameters default_parameters(R"({
+            "alpha_bossak"       : -0.3,
+            "update_acceleration": true,
+            "update_displacement": false,
+        })");
+        rParameters.ValidateAndAssignDefaults(default_parameters);
+
+        ResidualBasedBossakVelocityScheme(rParameters["alpha_bossak"].GetDouble(),
```

Comment:
You can't call the constructor like this. You are basically constructing a temporary and throwing it away. Your member data will not be initialized.

Reply:
this constructor is removed